riscv/triggers.cc

```diff
@@ -461,6 +518,19 @@ bool module_t::tdata3_write(unsigned index, const reg_t val) noexcept
   return true;
 }
 
+bool module_t::icount_enabled()
+{
+  if (check_icount_enabled) {
+    auto xlen = proc->get_xlen();
+    for (auto trigger: triggers) {
+      if (get_field(trigger->tdata1_read(proc), CSR_TDATA1_TYPE(xlen)) == CSR_TDATA1_TYPE_ICOUNT)
```

Comment:
Shouldn't this also consider other fields (such as `count > 0`)? If `type` is 3 but there's no `pending` and zero `count` then the trigger can't possibly fire and doesn't need to decrement.

Reply:
Actually I think this would best be done as a method on `trigger_t::icount_check_needed()` and then you can check if any trigger responds `true` to this method.